config.go

```diff
@@ -33,6 +33,17 @@ type Config struct {
 			Config *tls.Config
 		}
 
+		// SASL based authentication with broker. While there are multiple SASL authentication methods
+		// the current implementation is limited to plaintext (SASL/PLAIN) authentication
```

Comment:
Thanks for the comments.
With regard to enforcing use of TLS (not https) for SASL, I thought I would
keep the auth separate from the encryption to give people a choice should
the opt to use non tls transports with password authentication (as a
framework, it seemed harsh to enforce policies/preferences on the user).

Will address your other comments and send out a new PR soon.

On Thu, Apr 28, 2016 at 3:46 PM Evan Huus <notifications@github.com> wrote:

> In config.go
> <https://github.com/Shopify/sarama/pull/648#discussion_r61490796>:
>
> > @@ -33,6 +33,17 @@ type Config struct {
> >  			Config *tls.Config
> >  		}
> >
> > +		// SASL based authentication with broker. While there are multiple SASL authentication methods
> > +		// the current implementation is limited to plaintext (SASL/PLAIN) authentication
>
> since we currently only support plain-text, should we enforce that SASL is
> only used over HTTPS to avoid leaking credentials on the wire?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/sarama/pull/648/files/3834ba1ce06295021c65e824e3aa9337caf5f03f#r61490796>
>
-- 

~shriram